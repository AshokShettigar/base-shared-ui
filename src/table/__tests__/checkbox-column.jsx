import { shallow } from 'enzyme';
import CheckBoxColumn from 'platform-common-ui/table/checkbox-column';

describe('table/checkbox-column', function() {
    const onClick = () => true;

    it('should render the component', function() {
        const wrapper = shallow(<CheckBoxColumn onClick={onClick} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with is all selected flag', function() {
        const wrapper = shallow(<CheckBoxColumn isAllSelected onClick={onClick} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should test onClick on the component', function(done) {
        const onClickHandler = () => {
            expect(true).toBeTruthy();
            done();
        };
        const wrapper = shallow(<CheckBoxColumn onClick={onClickHandler} />);
        wrapper.simulate('click');
    });

    it('should render with custom id', function() {
        const wrapper = shallow(<CheckBoxColumn id="new_id" onClick={onClick} />);
        expect(wrapper).toMatchSnapshot();
    });
});
