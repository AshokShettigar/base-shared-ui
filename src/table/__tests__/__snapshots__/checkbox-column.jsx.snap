// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table/checkbox-column should render the component 1`] = `
<TableColumn
  id="checkbox_col"
  isSortable={false}
  onClick={[Function]}
  width="38px"
>
  <Icon
    className="cn-table__header-icon"
    icon="check-none"
    size="extra-small"
  />
</TableColumn>
`;

exports[`table/checkbox-column should render the component with is all selected flag 1`] = `
<TableColumn
  id="checkbox_col"
  isSortable={false}
  onClick={[Function]}
  width="38px"
>
  <Icon
    className="cn-table__header-icon"
    icon="check-all"
    size="extra-small"
  />
</TableColumn>
`;

exports[`table/checkbox-column should render with custom id 1`] = `
<TableColumn
  id="new_id"
  isSortable={false}
  onClick={[Function]}
  width="38px"
>
  <Icon
    className="cn-table__header-icon"
    icon="check-none"
    size="extra-small"
  />
</TableColumn>
`;
